packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -52,7 +64,7 @@ export class TableLayout<T> extends ListLayout<T> {
     let remainingSpace = this.virtualizer.visibleRect.width;
     for (let column of this.collection.columns) {
       let props = column.props as ColumnProps<T>;
-      let width = props.width ?? props.defaultWidth;
+      let width = props.width ?? this.getDefaultWidth(props);
```

Comment:
i think we can leave the check for props.defaultWidth here, it's not really part of the getDefaultWidth function which i guess would be more of a get implementation default width as opposed to a get user defined default width